modules/matcher.py

```diff
@@ -113,85 +122,97 @@ def cert_preprocess(df: pd.DataFrame, cert_feature: str) -> pd.DataFrame:
 ## Matching
 
 
-def direct_match(
-    feature_df: pd.DataFrame,
-    feature: str,
-    comparison_df: pd.DataFrame,
-    indicator=INDICATOR,
-) -> pd.DataFrame:
-    # filter out invalid data
-    feature_df = basic_preprocess(feature_df, indicator)
-    comparison_df = basic_preprocess(comparison_df, indicator)
-    test_matches = pd.merge(feature_df, comparison_df, how="inner", on=indicator)
+def direct_match(feature_df: pd.DataFrame, comparison_df: pd.DataFrame) -> pd.DataFrame:
+    test_matches = pd.merge(feature_df, comparison_df, how="inner", on=INDICATOR)
     # deduplicating
     matches = test_matches[test_matches.domain_name_x < test_matches.domain_name_y]
-    # note this throws a false positive SettingWithCopyWarning; the behavior is OK
-    matches[MATCH_TYPE] = feature
-    matches = matches.rename(columns={indicator: MATCH_VALUE})
+    matches = matches.rename(columns={INDICATOR: MATCH_VALUE})
     return matches.reset_index(drop=True)
 
-# TODO: Add a partial string match function
+def match_with_threshold(
+        feature_df,
+        comparison_df, 
+        match_function,
+        threshold
+):    
```

Comment:
this function handles the comparison of the two dataframes, matching, and thresholding. This way the other functions only have to handle the definition of the `match_function`